tools/triage/add_triage_signature.py

```diff
@@ -2626,60 +2627,119 @@ def _process_ticket(self, url, issue_key):
             self._update_triaging_ticket(report)
 
 
-class MissingOSTreePivot(ErrorSignature):
+class OSTreeCommitMismatch(ErrorSignature):
     """
     This signature looks for missing pivot URL in rpm-ostree status of the control-plane hosts
     """
 
-    PIVOT_URL_PATTERN = re.compile(r"pivot://.*")
+    # machine_os_content tag name has been changing overtime, the first ones in
+    # the list have preference (since machine-os-content still exists in the
+    # release image)
+    machine_os_content_names = ["rhel-coreos", "rhel-coreos-8", "machine-os-content"]
+    # The name of the labels also have been changing
+    ostree_commit_labels = ["ostree.commit", "com.coreos.ostree-commit"]
 
     def __init__(self, *args, **kwargs):
+        self.pull_secret = kwargs["pull_secret"]
         super().__init__(
             *args,
             **kwargs,
-            function_impact_label="missing_pivot_ostree",
-            comment_identifying_string="h1. MCO didn't pivot some hosts",
+            function_impact_label="ostree_commit_mismatch",
+            comment_identifying_string="h1. ostree commitid does not match the one in openshift release image",
         )
 
+    def get_os_content_image(self, release_image):
+        command = ["oc", "adm", "release", "info", release_image, "-ojson"]
+        if self.pull_secret:
+            command += ["--registry-config", self.pull_secret]
+        release_json = json.loads(subprocess.check_output(command))
+        tags = release_json["references"]["spec"]["tags"]
+        try:
+            # Get the first tag present in the release image that matches the machine_os_content list
+            # Order is important, since machine-os-content tag might be present on release_images where the correct one is rhel-coreos for example
+            image_name = next(name for name in self.machine_os_content_names if name in [tag["name"] for tag in tags])
+            # And return the image for that tag
+            return next(tag["from"]["name"] for tag in tags if tag["name"] == image_name)
+        except StopIteration:
+            return None
```

Comment:
maybe a regular ``for`` loop over ``tags`` will be clearer here, instead of usage of ``next`` and the stoppage at ``StopIteration``
any reason it won't work?